pkg/kubelet/status/manager.go

```diff
@@ -320,28 +320,28 @@ func (m *manager) RemoveOrphanedStatuses(podUIDs map[types.UID]bool) {
 // syncBatch syncs pods statuses with the apiserver.
 func (m *manager) syncBatch() {
 	var updatedStatuses []podStatusSyncRequest
-	podToMirror, mirrorToPod := m.podManager.GetUIDTranslations()
 	func() { // Critical section
 		m.podStatusesLock.RLock()
 		defer m.podStatusesLock.RUnlock()
 
 		// Clean up orphaned versions.
 		for uid := range m.apiStatusVersions {
-			_, hasPod := m.podStatuses[uid]
-			_, hasMirror := mirrorToPod[uid]
-			if !hasPod && !hasMirror {
+			// NOTE(random-liu): If a static pod doesn't have podStatuses, but has mirror pod,
+			// should we still keep it?
+			if _, hasPod := m.podStatuses[m.podManager.TranslatePodUID(uid)]; !hasPod {
 				delete(m.apiStatusVersions, uid)
 			}
 		}
 
 		for uid, status := range m.podStatuses {
 			syncedUID := uid
-			if mirrorUID, ok := podToMirror[uid]; ok {
-				syncedUID = mirrorUID
+			// NOTE(random-liu): Is this the same with the old implementation
+			if mirrorPod := m.podManager.GetMirrorPod(status.podName, status.podNamespace); mirrorPod != nil {
```

Comment:
I think @timstclair used `podToMirror, mirrorToPod := m.podManager.GetUIDTranslations()` at the beginning of this function to avoid accessing the pod manager while holding the `m.podStatusesLock` lock. Can we reuse the original implementation?